blog to share update on the stateful guidance for AKS highlighting Kakfa, Azure Container Storage and Terraform. (#4995)
Azure/AKS: blog/_posts/2025-05-28-stateful-guidance-update.md

## Introducing Guidance for Deploying Apache Kafka on AKS with the Strimzi Operator

Apache Kafka is an open source distributed event streaming platform designed to handle high-volume, high-throughput, and real-time streaming data. It is widely used by thousands of companies for mission-critical applications but managing and scaling Kafka clusters on Kubernetes can be challenging. [Strimzi](https://github.com/strimzi/strimzi-kafka-operator) simplifies the deployment and management of Kafka on Kubernetes by providing a set of Kubernetes Operators and container images that automate complex Kafka operational tasks. 

The [Kafka on Azure Kubernetes Service (AKS)](https://learn.microsoft.com/en-us/azure/aks/kafka-infrastructure?pivots=azure-cli) guide covers essential storage and compute considerations, ensuring you Kafka deployment meets your needs. Additionally, we provide guidance for tuning the Java Virtual Machine (JVM), which is critical for optimal Kafka broker and controller performance.

> In addition to basic installation steps, the Kafka guide provides best practice recommendations for configuring networking, monitoring, and  managed identity.

Reviewer: ```suggestion
On top of basic installation steps, the Kafka guide provides best practice recommendations for configuring networking, monitoring, and  managed identity.
```

Nit. Changing since "additionally" was used previously, avoid repetition.

## Updates to our current Guidance 

In addition to writing new guidance, we have reviewed our existing portfolio of guides to ensure they are accurate and up to date. Also, as new services and features have been developed, we have provided guidance to help you seamlessly integrate them with your stateful applications. 

## Azure Container Storage integration 

[Azure Container Storage](https://learn.microsoft.com/en-us/azure/storage/container-storage/container-storage-introduction) is a volume management, deployment, and orchestration service built for Kubernetes and based on [OpenEBS](https://openebs.io/). Azure Container Storage offers your clusters a significant advantage by effortlessly provisioning high-performance persistent volumes on ephemeral local NVMe drives, surpassing typical CSI drivers. 

> We have revised our [PostgreSQL guidance](https://learn.microsoft.com/en-us/azure/aks/create-postgresql-ha?tabs=acstor%2Chelm) and [Kafka guidance](https://learn.microsoft.com/en-us/azure/aks/kafka-overview) to include Azure Container Storage alongside other storage configuration options, so you can make an informed choice on what best suits your objectives. For the most durable data resiliency, you can use the Azure Disks CSI driver with Premium SSD disks which provide [zone-redundant storage](https://learn.microsoft.com/en-us/azure/storage/common/storage-redundancy#redundancy-in-the-primary-region) resiliency to your PostgreSQL deployment. For the best cost savings at scale, you can use [Premium SSD v2 disks](https://learn.microsoft.com/en-us/azure/virtual-machines/disks-deploy-premium-v2?tabs=azure-cli) which can let you set disk capacity independently from performance settings. For maximal performance, Azure Container Storage and ephemeral disks can provide the extremely low sub-millisecond latency and high input/output operations per second (IOPS) that transactional database workloads benefit from.  

Reviewer: ```suggestion
We have revised our [PostgreSQL guidance](https://learn.microsoft.com/en-us/azure/aks/create-postgresql-ha?tabs=acstor%2Chelm) and [Kafka guidance](https://learn.microsoft.com/en-us/azure/aks/kafka-overview) to include Azure Container Storage alongside other storage configuration options, so you can make an informed choice on what best suits your objectives. 
- For the most durable data resiliency, you can use the Azure Disks CSI driver with Premium SSD disks which provide [zone-redundant storage](https://learn.microsoft.com/en-us/azure/storage/common/storage-redundancy#redundancy-in-the-primary-region) resiliency to your PostgreSQL deployment. 
- For the best cost savings at scale, you can use [Premium SSD v2 disks](https://learn.microsoft.com/en-us/azure/virtual-machines/disks-deploy-premium-v2?tabs=azure-cli) which can let you set disk capacity independently from performance settings. 
- For maximum performance, Azure Container Storage and ephemeral disks can provide the extremely low sub-millisecond latency and high input/output operations per second (IOPS) that transactional database workloads benefit from.  
```
Suggest format change to help w/ readability.

---
title: "What's New?! Guidance Updates for Stateful Workloads on AKS"
description: "Learn how to set up an AKS cluster, deploy Kafka, integrate Azure Container Storage and explore the other stateful workloads running on AKS with Terraform or AzCLI."
date: 2025-05-28
author: Colin Mixon
categories: general
---
## Helping you deploy on AKS

> Building on our initial announcement for [Deploying Open Source Software on Azure](https://techcommunity.microsoft.com/blog/linuxandopensourceblog/deploying-open-source-software-on-azure-new-guides-for-aks-and-vms/4264602) we and the Customer Experience team are excited to announce we have expanded our library of techncial best practice deployment guides for stateful workloads on AKS. We have developed a comprehensive guide for deploying Kafka on AKS, updated our Postgres guidance with additional storage considerations for data resiliency, performance or costs, and added Terraform templates to our Mongo DB and Valkey guides for automated deployments. 

Reviewer 2: ```suggestion
Building on our initial announcement for [Deploying Open Source Software on Azure](https://techcommunity.microsoft.com/blog/linuxandopensourceblog/deploying-open-source-software-on-azure-new-guides-for-aks-and-vms/4264602) Azure is excited to announce we have expanded our library of technical best practice deployment guides for stateful workloads on AKS. We have developed a comprehensive guide for deploying Kafka on AKS, updated our Postgres guidance with additional storage considerations for data resiliency, performance or costs, and added Terraform templates to our Mongo DB and Valkey guides for automated deployments. 
```
